packages/core/src/cli/commands.ts

```diff
@@ -41,10 +45,21 @@ export function runCli(): void {
 
   buildCommand
     .option('-w --watch', 'turn on watch mode, watch for changes and rebuild')
+    .option('--clean', 'clean the output directory before building')
     .description('build the library for production')
     .action(async (options: BuildOptions) => {
       try {
         const rslibConfig = await loadConfig(options.config, options.envMode);
+
+        if (options.clean) {
+          const distDir = rslibConfig.output?.distPath?.root ?? OUTPUT_DIST_DIR;
```

Comment:
It's raw Rsbuild config, we should get `distDir` from final config.

Reply:
Rsbuild already support clean [output.cleanDistPath](https://rsbuild.dev/zh/config/output/clean-dist-path#outputcleandistpath), may be you should implemeny clean feature in plugin-dts, to add an option of clean, and in default rslib core logic, when we use builtin plugin-dts, pass `clean: true` to it.

Reply:
The issue was written early and may have some problems now.

Reply:
Would it be confusing to include the `clean` option in `plugin-dts`?

Reply:
Because `clean` is a global build option, not `plugin-dts`.

Reply:
We hope `rsbuild-plugin-dts` can be used seprately in a Rsbuild project, maybe we can inherit `output.cleanDistPath` options of Rsbuild and use its value to do DTS output clean?

Reply:
Maybe I misunderstood. Do we need to clear `DTS` output instead of `dist` output before building?

Reply:
Js output will be cleaned according to Rsbuild `output.cleanDistPath` by default, the DTS output should align with this behaviour. Maybe we don't need a cli option to do that, just in built-in logic.

Reply:
> Maybe we don't need a cli option to do that, just in built-in logic.

Agree

Reply:
However, `Rsbuild` seems to default to clear `dist`. What is the significance of doing so? Does `plugin-dts` not support default clear?

Reply:
The `dist` maybe dirty in local development, clean behavihour can ensure the new output will not be influenced by existed output.

Reply:
`plugin-dts` have not supported it yet.

We should clean:
- bundleless mode: the tsc emit path
- bundle mode: the tsc emit temp path `.rslib/declarations` and api-extractor emit path